docs/docs/01-ibc/06-channel-upgrades.md

```diff
@@ -11,6 +11,27 @@ slug: /ibc/channel-upgrades
 Learn how to upgrade existing IBC channels.
 :::
 
+## Channel State and Packet Flushing
+
+`FLUSHING` and `FLUSHCOMPLETE` are additional states which have been added to enable the upgrade feature.
+
+This is found in the channel state, on the `ChannelEnd` interface:
+
+```interface ChannelEnd {
+  state: ChannelState
+  ordering: ChannelOrder
+  counterpartyPortIdentifier: Identifier
+  counterpartyChannelIdentifier: Identifier
+  connectionHops: [Identifier]
+  version: string
+  upgradeSequence: uint64
+}
+```
+
+`startFlushing` is the specific method which is called in `ChanUpgradeTry` and `ChanUpgradeAck` to update the state on the channel end. This will set the timeout on the upgrade and update the channel state to `FLUSHING` which will block the upgrade from continuing until all in-flight packets have been flushed. The state will change to `FLUSHCOMPLETE` once there are no in-flight packets left and the channel end is ready to move to `OPEN`. This flush state will also have an impact on how a channel ugrade can be cancelled, as detailed below.
```

Comment:
Should we mention that when a channel end moves to `FLUSHING` packet sending becomes blocked?

Reply:
`This will set the timeout on the upgrade` maybe we could be here a bit more explicit: this sets the upgrade timeout for the counterparty (i.e. the timeout before which the counterparty chain must move from `FLUSHING` to `FLUSHCOMPLETE`; if it doesn't then the chain will cancel the upgrade and write an error receipt). The timeout is a relative time duration in nanoseconds that can be changed with `MsgUpdateParams` and by default is 10 minutes (I think this is the recommendation that relayers have shared with Susannah).

Reply:
i think i have the mention for blocking packet send already in the paragraph, ill update the timeout language!